app/_components/HamburgerMenu.tsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+import { useEffect, useRef, useState } from "react";
+import Link from "next/link";
+import Image from "next/image";
+import { ExitCrossFC, HamburgerMenuButton } from "../_atoms/Icons";
+
+const linkGroups = [
+  [
+    { href: "/", label: "Home" },
+    { href: "/about", label: "About Us" },
+    { href: "/contactus", label: "Contact Us" },
+  ],
+  [
+    { href: "/treatments", label: "Treatments" },
+    { href: "/clinics", label: "Clinics" },
+    { href: "/locations", label: "Locations" },
+    { href: "/blogs", label: "Blogs" },
+  ],
+  [
+    { href: "/faqs", label: "FAQs" },
+    { href: "/sitemap", label: "Sitemap" },
+  ],
+  [
+    { href: "/termsandconditions", label: "Terms & Conditions" },
+    { href: "/privacypolicy", label: "Privacy Policy" },
+  ],
+];
+
+const HamburgerMenu: React.FC = () => {
+  const [menuOpen, setMenuOpen] = useState<boolean>(false);
+  const menuRef = useRef<HTMLDivElement | null>(null);
+
+  const toggleMenu = () => setMenuOpen((prev) => !prev);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (menuRef.current && !menuRef.current.contains(event.target as Node)) {
+      setMenuOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  useEffect(() => {
+    document.body.classList.toggle("overflow-hidden", menuOpen);
+  }, [menuOpen]);
+
+  const renderLinks = (links: { href: string; label: string }[]) => (
+    <div className="pt-4">
+      {links.map(({ href, label }) => (
+        <Link href={href} key={href}>
+          <li
+            className="py-3 text-primary font-normal leading-4 transition hover:underline hover:rounded-l hover:bg-bgWhite hover:rounded-lg cursor-pointer"
+            onClick={() => setMenuOpen(false)}
+          >
+            {label}
+          </li>
+        </Link>
+      ))}
+    </div>
+  );
+
+  return (
+    <div className="flex items-center gap-4 relative z-10">
+      <button
+        onClick={toggleMenu}
+        className="block rounded p-2.5 text-accent transition hover:bg-bgWhite md:hidden"
+      >
+        <span className="sr-only">Toggle menu</span>
+        <HamburgerMenuButton />
+      </button>
+      <nav
+        ref={menuRef}
+        className={`fixed right-0 top-0 bottom-0 w-[85%] bg-white rounded-xl sm:hidden h-screen p-6 ease-in duration-300 z-100 overflow-hidden ${
+          menuOpen ? "fixed" : "hidden"
+        }`}
+      >
+        <div className="flex w-[60%] mt-8 mb-5 items-center justify-between">
+          <Image
+            src="/small-logo.png"
+            alt="company logo"
+            width={100}
+            height={30}
+            className="mx-1 cursor-pointer"
+          />
+          <div onClick={toggleMenu} className="cursor-pointer mr-5">
+            <ExitCrossFC />
```

Comment:
Could you adjust the icon's position (ExitCrossFC) slightly to the right to match the Figma design?